applications/SwimmingDEMApplication/custom_utilities/L2_error_projection_utility.h

```diff
@@ -73,32 +86,27 @@ double GetL2VectorProjection(ModelPart& r_model_part)
             for (unsigned int j = 0; j < NumNodes; ++j){
                 error_x += rGeom[j].FastGetSolutionStepValue(ERROR_X) * N[j];
                 error_y += rGeom[j].FastGetSolutionStepValue(ERROR_Y) * N[j];
+                error_z += 0.0;
                 if (dim == 3) error_z += rGeom[j].FastGetSolutionStepValue(ERROR_Z) * N[j];
             }
-            error.push_back(error_x);
-            error.push_back(error_y);
-            if (dim == 3) error.push_back(error_z);
-            // for (unsigned int d = 0; d < dim; ++d){
-            //     scalar_product[d] = error[d];
-            // }
-            result += std::pow(SWIMMING_MODULUS_3(error),2) * DetJ[gss] * IntegrationPoints[gss].Weight();
-            error.clear();
+            modulus = std::sqrt(std::pow(error_x,2)+std::pow(error_y,2)+std::pow(error_z,2));
+            result += std::pow(modulus,2) * DetJ[gss] * IntegrationPoints[gss].Weight();
```

Comment:
you are squaring a square root here... Please avoid it. Also look for a better name than 'interpolator'. Something that is descriptive of what it really is (something to do with the error).